src/basic/log.h

```diff
@@ -9,6 +9,12 @@
 
 #include "macro.h"
 
+/* The set of units that are logged about with log_syntax(). We use a global variable because those errors
+ * are usually ignored by the caller, and this way we can know how many issues (if any) were detected
+ * without refactoring all the callers. */
+typedef struct Set Set;
```

Comment:
I actually suggested this - including set.h here makes gcc explode with 4 pages of errors, nested deeply across 4/5 of our headers... and this is a pre-existing problem, if you try adding it on main it will happen regardless of this PR:

```
In file included from ../src/basic/gcrypt-util.h:12,
                 from ../src/basic/gcrypt-util.c:5:
../src/basic/string-util.h: In function ‘in_charset’:
../src/basic/macro.h:266:25: error: implicit declaration of function ‘log_assert_failed’ [-Werror=implicit-function-declaration]
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                         ^~~~~~~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
../src/basic/macro.h:266:25: warning: nested extern declaration of ‘log_assert_failed’ [-Wnested-externs]
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                         ^~~~~~~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
../src/basic/macro.h:266:52: note: each undeclared identifier is reported only once for each function it appears in
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
In file included from ../src/basic/siphash24.h:10,
                 from ../src/basic/hash-funcs.h:6,
                 from ../src/basic/hashmap.h:8,
                 from ../src/basic/set.h:5,
                 from ../src/basic/log.h:11,
                 from ../src/basic/macro.h:486,
                 from ../src/basic/gcrypt-util.h:12,
                 from ../src/basic/gcrypt-util.c:5:
../src/basic/string-util.h: In function ‘free_and_strdup_warn’:
../src/basic/string-util.h:172:24: error: implicit declaration of function ‘log_oom’ [-Werror=implicit-function-declaration]
                 return log_oom();
                        ^~~~~~~
../src/basic/string-util.h:172:24: warning: nested extern declaration of ‘log_oom’ [-Wnested-externs]
In file included from ../src/basic/gcrypt-util.h:12,
                 from ../src/basic/gcrypt-util.c:5:
../src/basic/string-util.h: In function ‘memory_startswith’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:194:9: note: in expansion of macro ‘assert’
         assert(token);
         ^~~~~~
../src/basic/string-util.h: In function ‘memory_startswith_no_case’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:212:9: note: in expansion of macro ‘assert’
         assert(token);
         ^~~~~~
../src/basic/util.h: In function ‘log2i’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/util.h:42:9: note: in expansion of macro ‘assert’
         assert(x > 0);
         ^~~~~~
../src/basic/util.h: In function ‘log2u’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/util.h:48:9: note: in expansion of macro ‘assert’
         assert(x > 0);
         ^~~~~~
../src/basic/util.h: In function ‘log2u_round_up’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/util.h:54:9: note: in expansion of macro ‘assert’
         assert(x > 0);
         ^~~~~~
In file included from ../src/basic/macro.h:486,
                 from ../src/basic/gcrypt-util.h:12,
                 from ../src/basic/gcrypt-util.c:5:
../src/basic/log.h: At top level:
../src/basic/log.h:170:17: warning: conflicting types for ‘log_assert_failed’
 _noreturn_ void log_assert_failed(
                 ^~~~~~~~~~~~~~~~~
In file included from ../src/basic/gcrypt-util.h:12,
                 from ../src/basic/gcrypt-util.c:5:
../src/basic/macro.h:266:25: note: previous implicit declaration of ‘log_assert_failed’ was here
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                         ^~~~~~~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
cc1: some warnings being treated as errors
[2/1685] Compiling C object src/libsystemd/libsystemd_static.a.p/sd-bus_bus-common-errors.c.o
FAILED: src/libsystemd/libsystemd_static.a.p/sd-bus_bus-common-errors.c.o 
ccache gcc -Isrc/libsystemd/libsystemd_static.a.p -Isrc/libsystemd -I../src/libsystemd -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I. -I.. -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=gnu99 -g -Wno-format-signedness -Wno-missing-field-initializers -Wno-unused-parameter -Wdate-time -Wendif-labels -Werror=format=2 -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=overflow -Werror=return-type -Werror=shift-count-overflow -Werror=shift-overflow=2 -Werror=undef -Wfloat-equal -Wimplicit-fallthrough=5 -Winit-self -Wlogical-op -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wsuggest-attribute=noreturn -Wwrite-strings -Wno-maybe-uninitialized -Wno-unused-result -Werror=missing-declarations -Werror=missing-prototypes -fdiagnostics-show-option -ffast-math -fno-common -fno-strict-aliasing -fstack-protector -fstack-protector-strong -fvisibility=hidden --param=ssp-buffer-size=4 -Werror=shadow -include config.h -ggdb3 -fPIC -pthread -fvisibility=default -MD -MQ src/libsystemd/libsystemd_static.a.p/sd-bus_bus-common-errors.c.o -MF src/libsystemd/libsystemd_static.a.p/sd-bus_bus-common-errors.c.o.d -o src/libsystemd/libsystemd_static.a.p/sd-bus_bus-common-errors.c.o -c ../src/libsystemd/sd-bus/bus-common-errors.c
In file included from ../src/libsystemd/sd-bus/bus-error.h:8,
                 from ../src/libsystemd/sd-bus/bus-common-errors.h:4,
                 from ../src/libsystemd/sd-bus/bus-common-errors.c:7:
../src/basic/string-util.h: In function ‘in_charset’:
../src/basic/macro.h:266:25: error: implicit declaration of function ‘log_assert_failed’ [-Werror=implicit-function-declaration]
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                         ^~~~~~~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
../src/basic/macro.h:266:25: warning: nested extern declaration of ‘log_assert_failed’ [-Wnested-externs]
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                         ^~~~~~~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
../src/basic/macro.h:266:52: note: each undeclared identifier is reported only once for each function it appears in
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
In file included from ../src/basic/siphash24.h:10,
                 from ../src/basic/hash-funcs.h:6,
                 from ../src/basic/hashmap.h:8,
                 from ../src/basic/set.h:5,
                 from ../src/basic/log.h:11,
                 from ../src/basic/macro.h:486,
                 from ../src/libsystemd/sd-bus/bus-error.h:8,
                 from ../src/libsystemd/sd-bus/bus-common-errors.h:4,
                 from ../src/libsystemd/sd-bus/bus-common-errors.c:7:
../src/basic/string-util.h: In function ‘free_and_strdup_warn’:
../src/basic/string-util.h:172:24: error: implicit declaration of function ‘log_oom’ [-Werror=implicit-function-declaration]
                 return log_oom();
                        ^~~~~~~
../src/basic/string-util.h:172:24: warning: nested extern declaration of ‘log_oom’ [-Wnested-externs]
In file included from ../src/libsystemd/sd-bus/bus-error.h:8,
                 from ../src/libsystemd/sd-bus/bus-common-errors.h:4,
                 from ../src/libsystemd/sd-bus/bus-common-errors.c:7:
../src/basic/string-util.h: In function ‘memory_startswith’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:194:9: note: in expansion of macro ‘assert’
         assert(token);
         ^~~~~~
../src/basic/string-util.h: In function ‘memory_startswith_no_case’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:212:9: note: in expansion of macro ‘assert’
         assert(token);
         ^~~~~~
../src/basic/util.h: In function ‘log2i’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/util.h:42:9: note: in expansion of macro ‘assert’
         assert(x > 0);
         ^~~~~~
../src/basic/util.h: In function ‘log2u’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/util.h:48:9: note: in expansion of macro ‘assert’
         assert(x > 0);
         ^~~~~~
../src/basic/util.h: In function ‘log2u_round_up’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/util.h:54:9: note: in expansion of macro ‘assert’
         assert(x > 0);
         ^~~~~~
In file included from ../src/basic/macro.h:486,
                 from ../src/libsystemd/sd-bus/bus-error.h:8,
                 from ../src/libsystemd/sd-bus/bus-common-errors.h:4,
                 from ../src/libsystemd/sd-bus/bus-common-errors.c:7:
../src/basic/log.h: At top level:
../src/basic/log.h:170:17: warning: conflicting types for ‘log_assert_failed’
 _noreturn_ void log_assert_failed(
                 ^~~~~~~~~~~~~~~~~
In file included from ../src/libsystemd/sd-bus/bus-error.h:8,
                 from ../src/libsystemd/sd-bus/bus-common-errors.h:4,
                 from ../src/libsystemd/sd-bus/bus-common-errors.c:7:
../src/basic/macro.h:266:25: note: previous implicit declaration of ‘log_assert_failed’ was here
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                         ^~~~~~~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
cc1: some warnings being treated as errors
[3/1685] Compiling C object src/shared/libsystemd-shared-249.a.p/generator.c.o
FAILED: src/shared/libsystemd-shared-249.a.p/generator.c.o 
ccache gcc -Isrc/shared/libsystemd-shared-249.a.p -Isrc/shared -I../src/shared -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I. -I.. -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -I/usr/include/blkid -I/usr/include/uuid -I/usr/include/libmount -I/usr/include/p11-kit-1 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=gnu99 -g -Wno-format-signedness -Wno-missing-field-initializers -Wno-unused-parameter -Wdate-time -Wendif-labels -Werror=format=2 -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=overflow -Werror=return-type -Werror=shift-count-overflow -Werror=shift-overflow=2 -Werror=undef -Wfloat-equal -Wimplicit-fallthrough=5 -Winit-self -Wlogical-op -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wsuggest-attribute=noreturn -Wwrite-strings -Wno-maybe-uninitialized -Wno-unused-result -Werror=missing-declarations -Werror=missing-prototypes -fdiagnostics-show-option -ffast-math -fno-common -fno-strict-aliasing -fstack-protector -fstack-protector-strong -fvisibility=hidden --param=ssp-buffer-size=4 -Werror=shadow -include config.h -ggdb3 -fPIC -pthread -fvisibility=default -MD -MQ src/shared/libsystemd-shared-249.a.p/generator.c.o -MF src/shared/libsystemd-shared-249.a.p/generator.c.o.d -o src/shared/libsystemd-shared-249.a.p/generator.c.o -c ../src/shared/generator.c
In file included from ../src/basic/alloc-util.h:9,
                 from ../src/shared/generator.c:6:
../src/basic/string-util.h: In function ‘in_charset’:
../src/basic/macro.h:266:25: error: implicit declaration of function ‘log_assert_failed’ [-Werror=implicit-function-declaration]
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                         ^~~~~~~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
../src/basic/macro.h:266:25: warning: nested extern declaration of ‘log_assert_failed’ [-Wnested-externs]
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                         ^~~~~~~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
../src/basic/macro.h:266:52: note: each undeclared identifier is reported only once for each function it appears in
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
In file included from ../src/basic/siphash24.h:10,
                 from ../src/basic/hash-funcs.h:6,
                 from ../src/basic/hashmap.h:8,
                 from ../src/basic/set.h:5,
                 from ../src/basic/log.h:11,
                 from ../src/basic/macro.h:486,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/shared/generator.c:6:
../src/basic/string-util.h: In function ‘free_and_strdup_warn’:
../src/basic/string-util.h:172:24: error: implicit declaration of function ‘log_oom’ [-Werror=implicit-function-declaration]
                 return log_oom();
                        ^~~~~~~
../src/basic/string-util.h:172:24: warning: nested extern declaration of ‘log_oom’ [-Wnested-externs]
In file included from ../src/basic/alloc-util.h:9,
                 from ../src/shared/generator.c:6:
../src/basic/string-util.h: In function ‘memory_startswith’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:194:9: note: in expansion of macro ‘assert’
         assert(token);
         ^~~~~~
../src/basic/string-util.h: In function ‘memory_startswith_no_case’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:212:9: note: in expansion of macro ‘assert’
         assert(token);
         ^~~~~~
In file included from ../src/basic/hashmap.h:8,
                 from ../src/basic/set.h:5,
                 from ../src/basic/log.h:11,
                 from ../src/basic/macro.h:486,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/shared/generator.c:6:
../src/basic/hash-funcs.h: At top level:
../src/basic/hash-funcs.h:14:9: error: unknown type name ‘free_func_t’
         free_func_t free_key;
         ^~~~~~~~~~~
../src/basic/hash-funcs.h:15:9: error: unknown type name ‘free_func_t’
         free_func_t free_value;
         ^~~~~~~~~~~
In file included from ../src/basic/alloc-util.h:9,
                 from ../src/shared/generator.c:6:
../src/basic/util.h: In function ‘log2i’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/util.h:42:9: note: in expansion of macro ‘assert’
         assert(x > 0);
         ^~~~~~
../src/basic/util.h: In function ‘log2u’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/util.h:48:9: note: in expansion of macro ‘assert’
         assert(x > 0);
         ^~~~~~
../src/basic/util.h: In function ‘log2u_round_up’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/util.h:54:9: note: in expansion of macro ‘assert’
         assert(x > 0);
         ^~~~~~
In file included from ../src/basic/set.h:5,
                 from ../src/basic/log.h:11,
                 from ../src/basic/macro.h:486,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/shared/generator.c:6:
../src/basic/hashmap.h: At top level:
../src/basic/hashmap.h:100:44: error: unknown type name ‘free_func_t’; did you mean ‘hash_func_t’?
 HashmapBase* _hashmap_free(HashmapBase *h, free_func_t default_free_key, free_func_t default_free_value);
                                            ^~~~~~~~~~~
                                            hash_func_t
../src/basic/hashmap.h:100:74: error: unknown type name ‘free_func_t’; did you mean ‘hash_func_t’?
 HashmapBase* _hashmap_free(HashmapBase *h, free_func_t default_free_key, free_func_t default_free_value);
                                                                          ^~~~~~~~~~~
                                                                          hash_func_t
../src/basic/hashmap.h: In function ‘hashmap_free’:
../src/basic/hashmap.h:102:24: error: implicit declaration of function ‘_hashmap_free’; did you mean ‘hashmap_free’? [-Werror=implicit-function-declaration]
         return (void*) _hashmap_free(HASHMAP_BASE(h), NULL, NULL);
                        ^~~~~~~~~~~~~
                        hashmap_free
../src/basic/hashmap.h:102:24: warning: nested extern declaration of ‘_hashmap_free’ [-Wnested-externs]
../src/basic/hashmap.h:102:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), NULL, NULL);
                ^
../src/basic/hashmap.h: In function ‘ordered_hashmap_free’:
../src/basic/hashmap.h:105:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), NULL, NULL);
                ^
../src/basic/hashmap.h: In function ‘hashmap_free_free’:
../src/basic/hashmap.h:109:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), NULL, free);
                ^
../src/basic/hashmap.h: In function ‘ordered_hashmap_free_free’:
../src/basic/hashmap.h:112:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), NULL, free);
                ^
../src/basic/hashmap.h: In function ‘hashmap_free_free_key’:
../src/basic/hashmap.h:116:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), free, NULL);
                ^
../src/basic/hashmap.h: In function ‘ordered_hashmap_free_free_key’:
../src/basic/hashmap.h:119:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), free, NULL);
                ^
../src/basic/hashmap.h: In function ‘hashmap_free_free_free’:
../src/basic/hashmap.h:123:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), free, free);
                ^
../src/basic/hashmap.h: In function ‘ordered_hashmap_free_free_free’:
../src/basic/hashmap.h:126:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), free, free);
                ^
In file included from ../src/basic/set.h:5,
                 from ../src/basic/log.h:11,
                 from ../src/basic/macro.h:486,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/shared/generator.c:6:
../src/basic/hashmap.h: At top level:
../src/basic/hashmap.h:289:37: error: unknown type name ‘free_func_t’; did you mean ‘hash_func_t’?
 void _hashmap_clear(HashmapBase *h, free_func_t default_free_key, free_func_t default_free_value);
                                     ^~~~~~~~~~~
                                     hash_func_t
../src/basic/hashmap.h:289:67: error: unknown type name ‘free_func_t’; did you mean ‘hash_func_t’?
 void _hashmap_clear(HashmapBase *h, free_func_t default_free_key, free_func_t default_free_value);
                                                                   ^~~~~~~~~~~
                                                                   hash_func_t
../src/basic/hashmap.h: In function ‘hashmap_clear’:
../src/basic/hashmap.h:291:9: error: implicit declaration of function ‘_hashmap_clear’; did you mean ‘hashmap_clear’? [-Werror=implicit-function-declaration]
         _hashmap_clear(HASHMAP_BASE(h), NULL, NULL);
         ^~~~~~~~~~~~~~
         hashmap_clear
../src/basic/hashmap.h:291:9: warning: nested extern declaration of ‘_hashmap_clear’ [-Wnested-externs]
In file included from ../src/basic/log.h:11,
                 from ../src/basic/macro.h:486,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/shared/generator.c:6:
../src/basic/set.h: In function ‘set_free’:
../src/basic/set.h:20:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (Set*) _hashmap_free(HASHMAP_BASE(s), NULL, NULL);
                ^
../src/basic/set.h: In function ‘set_free_free’:
../src/basic/set.h:24:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (Set*) _hashmap_free(HASHMAP_BASE(s), free, NULL);
                ^
In file included from ../src/basic/macro.h:486,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/shared/generator.c:6:
../src/basic/log.h: At top level:
../src/basic/log.h:170:17: warning: conflicting types for ‘log_assert_failed’
 _noreturn_ void log_assert_failed(
                 ^~~~~~~~~~~~~~~~~
In file included from ../src/basic/alloc-util.h:9,
                 from ../src/shared/generator.c:6:
../src/basic/macro.h:266:25: note: previous implicit declaration of ‘log_assert_failed’ was here
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                         ^~~~~~~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
cc1: some warnings being treated as errors
[4/1685] Compiling C object src/libsystemd/libsystemd_static.a.p/sd-journal_journal-send.c.o
FAILED: src/libsystemd/libsystemd_static.a.p/sd-journal_journal-send.c.o 
ccache gcc -Isrc/libsystemd/libsystemd_static.a.p -Isrc/libsystemd -I../src/libsystemd -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I. -I.. -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=gnu99 -g -Wno-format-signedness -Wno-missing-field-initializers -Wno-unused-parameter -Wdate-time -Wendif-labels -Werror=format=2 -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=overflow -Werror=return-type -Werror=shift-count-overflow -Werror=shift-overflow=2 -Werror=undef -Wfloat-equal -Wimplicit-fallthrough=5 -Winit-self -Wlogical-op -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wsuggest-attribute=noreturn -Wwrite-strings -Wno-maybe-uninitialized -Wno-unused-result -Werror=missing-declarations -Werror=missing-prototypes -fdiagnostics-show-option -ffast-math -fno-common -fno-strict-aliasing -fstack-protector -fstack-protector-strong -fvisibility=hidden --param=ssp-buffer-size=4 -Werror=shadow -include config.h -ggdb3 -fPIC -pthread -fvisibility=default -MD -MQ src/libsystemd/libsystemd_static.a.p/sd-journal_journal-send.c.o -MF src/libsystemd/libsystemd_static.a.p/sd-journal_journal-send.c.o.d -o src/libsystemd/libsystemd_static.a.p/sd-journal_journal-send.c.o -c ../src/libsystemd/sd-journal/journal-send.c
In file included from ../src/basic/alloc-util.h:9,
                 from ../src/libsystemd/sd-journal/journal-send.c:14:
../src/basic/string-util.h: In function ‘in_charset’:
../src/basic/macro.h:266:25: error: implicit declaration of function ‘log_assert_failed’ [-Werror=implicit-function-declaration]
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                         ^~~~~~~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
../src/basic/macro.h:266:25: warning: nested extern declaration of ‘log_assert_failed’ [-Wnested-externs]
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                         ^~~~~~~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
../src/basic/macro.h:266:52: note: each undeclared identifier is reported only once for each function it appears in
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
         ^~~~~~
In file included from ../src/basic/siphash24.h:10,
                 from ../src/basic/hash-funcs.h:6,
                 from ../src/basic/hashmap.h:8,
                 from ../src/basic/set.h:5,
                 from ../src/basic/log.h:11,
                 from ../src/basic/macro.h:486,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/libsystemd/sd-journal/journal-send.c:14:
../src/basic/string-util.h: In function ‘free_and_strdup_warn’:
../src/basic/string-util.h:172:24: error: implicit declaration of function ‘log_oom’ [-Werror=implicit-function-declaration]
                 return log_oom();
                        ^~~~~~~
../src/basic/string-util.h:172:24: warning: nested extern declaration of ‘log_oom’ [-Wnested-externs]
In file included from ../src/basic/alloc-util.h:9,
                 from ../src/libsystemd/sd-journal/journal-send.c:14:
../src/basic/string-util.h: In function ‘memory_startswith’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:194:9: note: in expansion of macro ‘assert’
         assert(token);
         ^~~~~~
../src/basic/string-util.h: In function ‘memory_startswith_no_case’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:212:9: note: in expansion of macro ‘assert’
         assert(token);
         ^~~~~~
In file included from ../src/basic/hashmap.h:8,
                 from ../src/basic/set.h:5,
                 from ../src/basic/log.h:11,
                 from ../src/basic/macro.h:486,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/libsystemd/sd-journal/journal-send.c:14:
../src/basic/hash-funcs.h: At top level:
../src/basic/hash-funcs.h:14:9: error: unknown type name ‘free_func_t’
         free_func_t free_key;
         ^~~~~~~~~~~
../src/basic/hash-funcs.h:15:9: error: unknown type name ‘free_func_t’
         free_func_t free_value;
         ^~~~~~~~~~~
In file included from ../src/basic/alloc-util.h:9,
                 from ../src/libsystemd/sd-journal/journal-send.c:14:
../src/basic/util.h: In function ‘log2i’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/util.h:42:9: note: in expansion of macro ‘assert’
         assert(x > 0);
         ^~~~~~
../src/basic/util.h: In function ‘log2u’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/util.h:48:9: note: in expansion of macro ‘assert’
         assert(x > 0);
         ^~~~~~
../src/basic/util.h: In function ‘log2u_round_up’:
../src/basic/macro.h:266:52: error: ‘PROJECT_FILE’ undeclared (first use in this function); did you mean ‘PROJECT_URL’?
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                                                    ^~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/util.h:54:9: note: in expansion of macro ‘assert’
         assert(x > 0);
         ^~~~~~
In file included from ../src/basic/set.h:5,
                 from ../src/basic/log.h:11,
                 from ../src/basic/macro.h:486,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/libsystemd/sd-journal/journal-send.c:14:
../src/basic/hashmap.h: At top level:
../src/basic/hashmap.h:100:44: error: unknown type name ‘free_func_t’; did you mean ‘hash_func_t’?
 HashmapBase* _hashmap_free(HashmapBase *h, free_func_t default_free_key, free_func_t default_free_value);
                                            ^~~~~~~~~~~
                                            hash_func_t
../src/basic/hashmap.h:100:74: error: unknown type name ‘free_func_t’; did you mean ‘hash_func_t’?
 HashmapBase* _hashmap_free(HashmapBase *h, free_func_t default_free_key, free_func_t default_free_value);
                                                                          ^~~~~~~~~~~
                                                                          hash_func_t
../src/basic/hashmap.h: In function ‘hashmap_free’:
../src/basic/hashmap.h:102:24: error: implicit declaration of function ‘_hashmap_free’; did you mean ‘hashmap_free’? [-Werror=implicit-function-declaration]
         return (void*) _hashmap_free(HASHMAP_BASE(h), NULL, NULL);
                        ^~~~~~~~~~~~~
                        hashmap_free
../src/basic/hashmap.h:102:24: warning: nested extern declaration of ‘_hashmap_free’ [-Wnested-externs]
../src/basic/hashmap.h:102:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), NULL, NULL);
                ^
../src/basic/hashmap.h: In function ‘ordered_hashmap_free’:
../src/basic/hashmap.h:105:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), NULL, NULL);
                ^
../src/basic/hashmap.h: In function ‘hashmap_free_free’:
../src/basic/hashmap.h:109:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), NULL, free);
                ^
../src/basic/hashmap.h: In function ‘ordered_hashmap_free_free’:
../src/basic/hashmap.h:112:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), NULL, free);
                ^
../src/basic/hashmap.h: In function ‘hashmap_free_free_key’:
../src/basic/hashmap.h:116:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), free, NULL);
                ^
../src/basic/hashmap.h: In function ‘ordered_hashmap_free_free_key’:
../src/basic/hashmap.h:119:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), free, NULL);
                ^
../src/basic/hashmap.h: In function ‘hashmap_free_free_free’:
../src/basic/hashmap.h:123:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), free, free);
                ^
../src/basic/hashmap.h: In function ‘ordered_hashmap_free_free_free’:
../src/basic/hashmap.h:126:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (void*) _hashmap_free(HASHMAP_BASE(h), free, free);
                ^
In file included from ../src/basic/set.h:5,
                 from ../src/basic/log.h:11,
                 from ../src/basic/macro.h:486,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/libsystemd/sd-journal/journal-send.c:14:
../src/basic/hashmap.h: At top level:
../src/basic/hashmap.h:289:37: error: unknown type name ‘free_func_t’; did you mean ‘hash_func_t’?
 void _hashmap_clear(HashmapBase *h, free_func_t default_free_key, free_func_t default_free_value);
                                     ^~~~~~~~~~~
                                     hash_func_t
../src/basic/hashmap.h:289:67: error: unknown type name ‘free_func_t’; did you mean ‘hash_func_t’?
 void _hashmap_clear(HashmapBase *h, free_func_t default_free_key, free_func_t default_free_value);
                                                                   ^~~~~~~~~~~
                                                                   hash_func_t
../src/basic/hashmap.h: In function ‘hashmap_clear’:
../src/basic/hashmap.h:291:9: error: implicit declaration of function ‘_hashmap_clear’; did you mean ‘hashmap_clear’? [-Werror=implicit-function-declaration]
         _hashmap_clear(HASHMAP_BASE(h), NULL, NULL);
         ^~~~~~~~~~~~~~
         hashmap_clear
../src/basic/hashmap.h:291:9: warning: nested extern declaration of ‘_hashmap_clear’ [-Wnested-externs]
In file included from ../src/basic/log.h:11,
                 from ../src/basic/macro.h:486,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/libsystemd/sd-journal/journal-send.c:14:
../src/basic/set.h: In function ‘set_free’:
../src/basic/set.h:20:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (Set*) _hashmap_free(HASHMAP_BASE(s), NULL, NULL);
                ^
../src/basic/set.h: In function ‘set_free_free’:
../src/basic/set.h:24:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         return (Set*) _hashmap_free(HASHMAP_BASE(s), free, NULL);
                ^
In file included from ../src/basic/macro.h:486,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/libsystemd/sd-journal/journal-send.c:14:
../src/basic/log.h: At top level:
../src/basic/log.h:170:17: warning: conflicting types for ‘log_assert_failed’
 _noreturn_ void log_assert_failed(
                 ^~~~~~~~~~~~~~~~~
In file included from ../src/basic/alloc-util.h:9,
                 from ../src/libsystemd/sd-journal/journal-send.c:14:
../src/basic/macro.h:266:25: note: previous implicit declaration of ‘log_assert_failed’ was here
                         log_assert_failed(message, PROJECT_FILE, __LINE__, __PRETTY_FUNCTION__); \
                         ^~~~~~~~~~~~~~~~~
../src/basic/macro.h:282:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^~~~~~~~~~~~~~~~~
../src/basic/string-util.h:127:9: note: in expansion of macro ‘assert’
         assert(s);
```

Reply:
@maanyagoenka please move this definition below the next two, and update the comment to say that we don't want to pull those headers to avoid conflicts due to cross-includes